core/src/commonJvmMain/kotlin/com/ustadmobile/core/network/containeruploader/ContainerUploader2.kt

```diff
@@ -100,31 +101,38 @@ actual class ContainerUploader2 actual constructor(
                     progressListener?.onProgress(
                         uploadSessionParams.startFrom + bytesUploaded, bytesToUpload)
                 }
+            }else{
+                bytesToUpload = 0
             }
         }catch(e: Exception) {
-            exception = e
             e.printStackTrace()
+            throw e
         }finally {
             //if we are here because we got canceled, read anything remainder to null so that the
             //write job will not get stuck
             pipeIn?.copyTo(NullOutputStream())
             pipeIn?.close()
             response?.closeQuietly()
+        }
 
-            try {
-                httpClient.get<Unit>("${endpoint.url}ContainerUpload2/${request.uploadUuid}/close")
-            }catch(e: Exception){
-                //do nothing
+        try {
```

Comment:
This can be moved inside the first try/catch - no need for a second one

Reply:
done